presto-client/src/main/java/com/facebook/presto/client/QueryResults.java

```diff
@@ -312,6 +314,8 @@ private static Object fixValue(TypeSignature signature, Object value)
             case INTERVAL_DAY_TO_SECOND:
             case DECIMAL:
                 return String.class.cast(value);
+            case CHAR:
+                return Strings.padEnd(String.class.cast(value), getCharTypeLength(signature), ' ');
```

Comment:
Discussed with @martint and he agreed that this isn't ideal. Lets move this to the server side and if we need a more efficient transport we can do that when a binary protocol is implemented (which is needed for more efficient transport of other types like BIGINT)